### PR TITLE
Dg update appendix

### PR DESCRIPTION
**Pull request by kevinchangjk**
Updated DG, particularly for:

- NFRs
- Glossary
- Manual testing for all commands

Also made changes to:
- Fix `index.md` image bug
- Remove destroying of object in `find` sequence diagram
- Random UG & DG changes

If there are any NFRs or Glossary terms or manual test instructions that you think should be added, please review and comment. I think it still is rather lacking.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -591,42 +633,322 @@ testers are expected to do more *exploratory* testing.
...
+   1. **Prerequisites**: For all of these test cases, ensure that there is at least 1 employee present in the database.
+      Additionally, take note of the total count of employees (you can see this by entering `list` and checking the largest index shown).
+
+   2. **Test Case**: `view 1`<br>
+      **Expected Outcome**: On the right side panel, the employee profile of the first listed person is displayed.
+
+   3. **Test Case**: `view 0`<br>
+      **Expected Outcome**: An error message appears, saying that the format of the command entered is incorrect.
+
+   4. **Test Case**: `view x`, where `x` is a number larger than the total count of employees<br>
```
Comment (prit3010) [inline docs/DeveloperGuide.md:687]: Case where employee is deleted

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -591,42 +633,322 @@ testers are expected to do more *exploratory* testing.
...
+      **Expected Outcome**: A new employee is displayed on the left panel, with the employee ID value of 1.
+
+### Adding an employee
+
+1. Adding an employee with all fields specified:
+
+   1. **Test Case**: `add n/John Doe j/Accountant d/Finance p/91234567 e/john@coydir.com a/Oak Street, Loudoun l/21 t/New hire`<br>
+      **Expected Outcome**: A new employee, with the name "John Doe", appears at the bottom of the list of employees on the left panel (you might need to scroll down to view it).
+
+2. Adding an employee with optional fields missing:
```
Comment (prit3010) [inline docs/DeveloperGuide.md:711]: Test case where field is not valid

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -591,42 +633,322 @@ testers are expected to do more *exploratory* testing.
...
+Peter Lim,99272758,peter@example.com,Accountant,Finance,Blk 30 Lorong 3 Serangoon Gardens #07-18,14,collegaues/friends
+Shawn Kok,99272758,shawn@example.com,Recruiter,Human Resources,Blk 30 Lorong 3 Serangoon Gardens #07-18,14,collegaues/friends
+Kevin Chang,99272758,kevin@example.com,Factory Worker,Production,Blk 30 Lorong 3 Serangoon Gardens #07-18,14,collegaues/friends
+Ng Shi Jun,99272758,shijun@example.com,Frontend Developer,Information Technology,Blk 30 Lorong 3 Serangoon Gardens #07-18,14,collegaues/friends
+Prittam Ravi,87438807,prittam@example.com,Chief Technology Officer,General Management,Blk 30 Lorong 3 Serangoon Gardens #07-18,14,collegaues/friends
+```
+
+1. Batch-adding multiple employees:
+
+   1. Prerequisites: Must not have used any of the names provided in `data/success.csv`.
```
Comment (prit3010) [inline docs/DeveloperGuide.md:821]: can add no commas in the fields within the CSV file

Comment (prit3010) [reply docs/DeveloperGuide.md:821]: Ordering of the columns need to follow the one provided

**Review by prit3010 — CHANGES_REQUESTED** [review]
Minor changes! Thank you

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (kevinchangjk) [reply docs/DeveloperGuide.md:687]: Noted.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (kevinchangjk) [reply docs/DeveloperGuide.md:711]: Okay, will add.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (kevinchangjk) [reply docs/DeveloperGuide.md:821]: Yep will add.

**Review by prit3010 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -591,42 +633,330 @@ testers are expected to do more *exploratory* testing.
...
+Therefore, we recommend that when performing a test case for an employee (with ID `x`), you first view the profile to easily observe the outcome.
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+1. Rating an employee's performance
 
-   1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+   1. **Test Case**: `rate id/x r/5`<br>
+      **Expected Outcome**: The "Performance" property of the employee is updated to "5".
+      On the "Performance History" graph, a point with value "5" is now present, with today's date labelled.
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:941]: node with value "5"

Comment (Kok-je) [reply docs/DeveloperGuide.md:941]: On the "Performance History" graph, there will be a new node with a rating value of "5", and a timestamp of today's date.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -591,42 +633,330 @@ testers are expected to do more *exploratory* testing.
...
+For the following tests, ensure that there is at least 1 employee, who has never been rated prior.
+If there are more than one, any of these employees can be used, and can be used for different test cases as well.
+In each of the test cases, we will refer to this employee's ID as `x`, for generality.
 
-1. Deleting a person while all persons are being shown
+**Note**: Changes are mostly visible only when viewing the employee profile.
+Therefore, we recommend that when performing a test case for an employee (with ID `x`), you first view the profile to easily observe the outcome.
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+1. Rating an employee's performance
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:937]: Should include the scenario where user rating is out of the acceptable range, example r/6

**Review by Kok-je — COMMENTED** [review]
Some minor changes otherwise lgtm

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (kevinchangjk) [reply docs/DeveloperGuide.md:941]: Okay, will change to that.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (kevinchangjk) [reply docs/DeveloperGuide.md:937]: Okay will add.

**Review by chao890 — APPROVED** [review]
LGTM

**Pull request merged**